Fix shared lib and pkg config files

Apteryks opened the pull request:
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

> /area build

> /area libscap-engine-gvisor

> /area libpman

> /area libsinsp

**What this PR does / why we need it**:

It sanitizes the generated pkg-config files of libscap.pc and libsinsp.pc, and add missing includes needed to build as shared libraries.

**Which issue(s) this PR fixes**:

Fixes #1820

**Special notes for your reviewer**:

Also see https://github.com/falcosecurity/libs/issues/1825, which is not addressed by this PR but can be easily worked around.

**Does this PR introduce a user-facing change?**:

No.

poiana commented:
Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1842#" title="Author self-approved">Apteryks</a>*
**Once this PR has been reviewed and has the lgtm label**, please assign leogr for approval. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["leogr"]} -->

poiana commented:
Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.

:memo: **Please follow instructions in the [contributing guide](https://github.com/falcosecurity/.github/blob/main/CONTRIBUTING.md) to update your commits with the DCO**

Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).

**The list of commits missing DCO signoff**:

- [9184dfe](https://github.com/falcosecurity/libs/commits/9184dfe3d1f3b94419c995f0b7dd1ba4dd031c64) userspace: Extend CFLAGS of libscap.pc and libsinsp.pc.
- [0e603e4](https://github.com/falcosecurity/libs/commits/0e603e43159c7e95fa1aba0caf68055b584acc72) build: Install the pman library and its header.
- [bf7c46a](https://github.com/falcosecurity/libs/commits/bf7c46a501a3ac6ec39772864c0ce8e47b87a666) libscap: Guard against install_lib_link_libraries-NOTFOUND
- [192681c](https://github.com/falcosecurity/libs/commits/192681cf4100641527f85ac9b42beaf8c5bb9fbf) build: Fix shared library build.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

poiana commented:
Welcome @Apteryks! It looks like this is your first PR to falcosecurity/libs 🎉

Apteryks commented:
I've now successfully built sysdig against a shared library falcosecurity-libs distinct package with this series, on GNU Guix.

FedeDP commented:
Hi! Thanks for this PR!
I will invoke our cmake experts for a review :) 
To me, changes look good btw!

cc @federico-sysdig @geraldcombs

federico-sysdig commented on userspace/libscap/engine/gvisor/CMakeLists.txt:
```diff
@@ -76,6 +76,14 @@ if (BUILD_SHARED_LIBS)
     add_dependencies(scap scap_engine_gvisor_o)
     target_sources(scap PRIVATE $<TARGET_OBJECTS:scap_engine_gvisor_o>)
+
+    target_include_directories(scap_engine_gvisor_o
+        PRIVATE
+        ${CMAKE_BINARY_DIR}
+        ${CMAKE_CURRENT_BINARY_DIR}
+        ${CMAKE_SOURCE_DIR}
+        ${CMAKE_SOURCE_DIR}/userspace
+    )
```
I'm not sure I'd use `target_include_directories` for bringing in the needed include paths. A `target_link_libraries` line would express the dependency in a better way. 
Is it because of the circular dependency mentioned above?
I'm not objecting to this if there's such a limitation, but let's consider this as a hack to be improved up in the future, perhaps solving the circular dependency (I know it's a harder problem, most likely).

Apteryks replied:
I tried that but it didn't have any effect.   Reading on it, it was because an OBJECT "library" is just an object, and is not linked.  So any link directives added to it are ignored by CMake, as far as I understand.

federico-sysdig commented on userspace/libsinsp/CMakeLists.txt:
```diff
@@ -169,12 +169,14 @@ target_link_libraries(sinsp
-	"${CURL_LIBRARIES}"
-	"${JSONCPP_LIB}"
-	"${RE2_LIB}"
+set(SINSP_PKGCONFIG_LIBRARIES)
+set(SINSP_PKGCONFIG_REQUIRES
+	$<$<BOOL:${JSONCPP_LIB}>:jsoncpp>
+)
+
+set(SINSP_PKGCONFIG_REQUIRES_PRIVATE
+	$<$<BOOL:${CURL_LIBRARIES}>:libcurl>
+	$<$<BOOL:${RE2_LIB}>:re2>
```
Ouch. Why this?
I know we don't always use `find_package` which would provide nice target aliases like `CURL::libcurl` and `re2::re2`, which bring on board all the properties a "consumer" of those libraries would need (include path, binary to link to, etc.). But why this? It seems like a workaround to give a target when the variable is not available. I'd address the issue upstream and ensure the variable has a value. Still inferior to having the standard target aliases for each library, but not relying on generator expression black magic. Please, forgive me for being direct on this. I do believe simplicity is key when dealing with CMake.

Apteryks replied:
It was a defensive technique to avoid adding a pkg-config module name in case a library was not found and this was not a fatal error.

Looking at it more closely, it seems this should never occur, so I've removed these safeguards, for better readability.

federico-sysdig reviewed (commented):
It's an interesting PR and there's excellent work on the corrections for the generated pkgconfig files.
I do believe that something else is desirable for the management of the dependent libraries other than the proposed generator expressions.

I'm curious; I have tried to implement the ability to integrate `falcosecurity/libs` in a client project through `find_package`, which is a better, more CMake-oriented, way to use a library. This is not to say that pkgconfig should be left behind, just another option. What are your thoughts on that?

geraldcombs commented on userspace/libsinsp/CMakeLists.txt:
```diff
@@ -306,41 +313,25 @@ if(NOT DEFINED SINSP_AGENT_CGROUP_MEM_PATH_ENV_VAR)
+
+Requires: libscap $<JOIN:${SINSP_PKGCONFIG_REQUIRES}, >
+Requires.private: $<JOIN:${SINSP_PKGCONFIG_REQUIRES_PRIVATE}, >
+Libs: -L\${libdir} -lsinsp $<JOIN:${SINSP_PKG_CONFIG_LIBS}, -l>
```
Did you mean SINSP_PKGCONFIG_LIBRARIES here?

Apteryks replied:
Good catch, though we would also need to guard against SINSP_PKGCONFIG_LIBRARIES being empty, perhaps?

Apteryks replied:
The reason I missed that is that I was using the wrong variable to start with (it should have been `SINSP_PKGCONFIG_LIBRARIES`).

Apteryks replied:
> Good catch, though we would also need to guard against SINSP_PKGCONFIG_LIBRARIES being empty, perhaps?

Done, with: `Libs: -L\${libdir} -lsinsp $<BOOL:${SINSP_PKGCONFIG_LIBRARIES}:-l$<JOIN:${SINSP_PKGCONFIG_LIBRARIES}, -l>>`

geraldcombs commented on userspace/libsinsp/CMakeLists.txt:
```diff
@@ -306,41 +313,25 @@ if(NOT DEFINED SINSP_AGENT_CGROUP_MEM_PATH_ENV_VAR)
+list(REMOVE_DUPLICATES SINSP_PKGCONFIG_REQUIRES)
+list(REMOVE_DUPLICATES SINSP_PKGCONFIG_REQUIRES_PRIVATE)
+
+file(GENERATE
```
Configuring libscap.pc and generating libsinsp.pc seems kind of odd. Shouldn't we create each file the same way?

Apteryks replied:
I've reverted to the old way, since the generator expressions could be reduced, limiting the need for file(GENERATE ...) (which is the only one supporting genex -- configure_file does not).

Apteryks commented:
> It's an interesting PR and there's excellent work on the corrections for the generated pkgconfig files. I do believe that something else is desirable for the management of the dependent libraries other than the proposed generator expressions.

I'll get to these in a bit.

> I'm curious; I have tried to implement the ability to integrate `falcosecurity/libs` in a client project through `find_package`, which is a better, more CMake-oriented, way to use a library. This is not to say that pkgconfig should be left behind, just another option. What are your thoughts on that?

My immediate thought on this is that a `FindFalcosecurityLibs.cmake` or similarly named module implementing the logic for`find_package` could be implemented via`pkg_check_modules`, with the obvious drawback that it adds a requirement on a `pkg-config` binary being available in the environment. Whatever the implementation detail chosen for an eventual CMake-based find_package module for this project, I think it can and should remain a distinct effort from this PR, which focuses on improving the pkg-config generated files :-).

federico-sysdig commented:
> Whatever the implementation detail chosen for an eventual CMake-based find_package module for this project, I think it can and should remain a distinct effort from this PR, which focuses on improving the pkg-config generated files :-).

Of course, I wasn't suggesting to change the scope of this PR.

geraldcombs commented:
The "build-shared-libs-macos-amd64" job is failing because CFlags in libscap.pc no longer includes `-I/opt/homebrew/include`:

```
In file included from /tmp/libs-test/include/falcosecurity/libsinsp/sinsp.h:45:
In file included from /tmp/libs-test/include/falcosecurity/libscap/scap.h:66:
/tmp/libs-test/include/falcosecurity/libscap/uthash_ext.h:24:10: fatal error: 'uthash.h' file not found
#include "uthash.h"
         ^~~~~~~~~~
```

Apteryks commented:
> The "build-shared-libs-macos-amd64" job is failing because CFlags in libscap.pc no longer includes `-I/opt/homebrew/include`:
> 
> ```
> In file included from /tmp/libs-test/include/falcosecurity/libsinsp/sinsp.h:45:
> In file included from /tmp/libs-test/include/falcosecurity/libscap/scap.h:66:
> /tmp/libs-test/include/falcosecurity/libscap/uthash_ext.h:24:10: fatal error: 'uthash.h' file not found
> #include "uthash.h"
>          ^~~~~~~~~~
> ```

The Cflags of the libscap.pc file shouldn't have changed; I've only added two new entries to them.  For example, on my machine, the old copy (master) looks like:
```
 $ cat ./libscap/libscap.pc 
prefix=${pcfiledir}/../..
libdir=${prefix}/lib64
includedir=${prefix}/include

Name: libscap
Description: lib for System CAPture
Version: 0.0.0

Libs: -L${libdir} -L/gnu/store/65iv1fc7z68cammywngh0dn5995ylbmk-profile/lib -L/gnu/store/65iv1fc7z68cammywngh0dn5995ylbmk-profile/lib -L/gnu/store/65iv1fc7z68cammywngh0dn5995ylbmk-profile/lib -L/gnu/store/65iv1fc7z68cammywngh0dn5995ylbmk-profile/lib -L/gnu/store/65iv1fc7z68cammywngh0dn5995ylbmk-profile/lib -lscap -lz -lprotobuf -ljsoncpp -lscap_engine_nodriver -lscap_engine_test_input -lscap_engine_source_plugin -lscap_engine_kmod -lscap_engine_bpf -lelf -lscap_engine_modern_bpf -lpman
Cflags: -I${includedir}/falcosecurity/libscap
```

With this change it now reads:
```
$ cat ../build/libscap/libscap.pc
prefix=/usr/local
libdir=${prefix}/lib64
includedir=${prefix}/include

Name: libscap
Description: lib for System CAPture
Version: 0.0.0

Requires: zlib
Libs: -L${libdir} -L{libdir}/falcosecurity/libscap  -lscap -lscap_engine_nodriver -lscap_engine_test_input -lscap_engine_source_plugin -lscap_engine_kmod -lscap_engine_bpf -lscap_engine_modern_bpf
Cflags: -I${includedir}/falcosecurity/libscap -I${includedir}/falcosecurity/driver -I${includedir}/falcosecurity
```

Probably this header was found via libsinsp.pc, which was capturing a lot of build-specific, non-installed directories, which I think shouldn't be baked in the generated .pc file.

Previously, it looked like:
```
$ cat ./libsinsp/libsinsp.pc 
prefix=${pcfiledir}/../..
libdir=${prefix}/lib64
includedir=${prefix}/include

Name: libsinsp
Description: lib for System INSPection
Version: 0.0.0

Requires: libscap
Libs: -L${libdir} -lsinsp -L/gnu/store/65iv1fc7z68cammywngh0dn5995ylbmk-profile/lib -lz -lcurl -ljsoncpp -lre2 -lcares -lgRPC::grpc++ -lgRPC::grpc -lgRPC::gpr -lprotobuf -lcares -lrt -lanl -lssl -lcrypto -ldl -lpthread
Cflags: -I${includedir}/falcosecurity/libsinsp -I/home/maxim/src/falcosecurity-libs -I/home/maxim/src/falcosecurity-libs/userspace -I/home/maxim/src/falcosecurity-libs/userspace/libscap -I/home/maxim/src/falcosecurity-libs/build_orig -I/home/maxim/src/falcosecurity-libs/build_orig/driver/src -I/gnu/store/65iv1fc7z68cammywngh0dn5995ylbmk-profile/include/tbb -I/gnu/store/65iv1fc7z68cammywngh0dn5995ylbmk-profile/include
```

Now, it looks like:
```
$ cat ../build/libsinsp/libsinsp.pc 
prefix=/usr/local
libdir=${prefix}/lib64
includedir=${prefix}/include

Name: libsinsp
Description: lib for System INSPection
Version: 0.0.0

Requires: libscap jsoncpp libcares gpr grpc grpc++ protobuf libcrypto libssl
Requires.private: libcurl re2 tbb
Libs: -L${libdir} -lsinsp -lrt -lanl -ldl -lpthread
Cflags: -I${includedir}/falcosecurity/libsinsp -I${includedir}/falcosecurity/driver -I${includedir}/falcosecurity
```

geraldcombs commented on userspace/libscap/libscap.pc.in:
```diff
@@ -1,10 +1,11 @@
-Cflags: -I${includedir}/@LIBS_PACKAGE_NAME@/libscap
+Requires: zlib
+Libs: -L${libdir} -L{libdir}/@LIBS_PACKAGE_NAME@/libscap @LIBSCAP_LINK_LIBDIRS_FLAGS@ @LIBSCAP_LINK_LIBRARIES_FLAGS@
+Cflags: -I${includedir}/@LIBS_PACKAGE_NAME@/libscap -I${includedir}/@LIBS_PACKAGE_NAME@/driver -I${includedir}/@LIBS_PACKAGE_NAME@
```
We should probably add `-I{$UTHASH_INCLUDE}` here in order to be more strictly compliant with the pkg-config specification.

It also looks like the ".../libscap" and ".../driver" entries aren't strictly needed as of 6f8cb4707. I suppose it's a question of how convenient we want to make things for people who might include "scap.h" instead of "userspace/scap.h".

Apteryks replied:
I've now added -I@UTHASH_INCLUDE@ to libscap.pc.in, thanks for pointing that.  About whether we should allow the includes to work without the libscap and libsinsp prefixes, I'll leave the call to you, but note that at least some test files still do:

```

userspace/libsinsp/test/classes/sinsp.cpp:21:#include <scap.h>
userspace/libsinsp/test/plugins/plugin_source.cpp:24:#include <scap.h>
userspace/libsinsp/test/plugins/syscall_async.cpp:26:#include <scap.h>
userspace/libsinsp/test/plugins/syscall_source.cpp:26:#include <scap.h>
```

as of current master.

geraldcombs commented on userspace/libsinsp/libsinsp.pc.in:
```diff
@@ -1,11 +1,12 @@
+Requires: libscap @LIBSINSP_REQUIRES@
+Requires.private: @LIBSINSP_REQUIRES_PRIVATE@
+Libs: -L${libdir} -lsinsp @LIBSINSP_LINK_FLAGS@
+Cflags: -I${includedir}/@LIBS_PACKAGE_NAME@/libsinsp -I${includedir}/@LIBS_PACKAGE_NAME@/driver -I${includedir}/@LIBS_PACKAGE_NAME@
```
The note above about ".../libscap" and ".../driver" applies here as well.

Apteryks replied:
I don't feel strongly enough to make the change, but feel free to edit these out if it doesn't cause any problem to running the test suite and building sysdig.

Apteryks replied:
(also see my other comment above about some files still relying on `#include <scap.h>`.

FedeDP commented:
/milestone 0.18.0

poiana commented:
Issues go stale after 90d of inactivity.

Mark the issue as fresh with `/remove-lifecycle stale`.

Stale issues rot after an additional 30d of inactivity and eventually close.

If this issue is safe to close now please do so with `/close`.

Provide feedback via https://github.com/falcosecurity/community.

/lifecycle stale

FedeDP commented:
/remove-lifecycle stale

FedeDP commented:
Hi! Can you rebase on latest master? 
Let's see if CI issues will disappear :D

poiana commented:
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1842#" title="Author self-approved">Apteryks</a>*
**Once this PR has been reviewed and has the lgtm label**, please assign [leogr](https://github.com/leogr) for approval. For more information see [the Kubernetes Code Review Process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process).

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":["leogr"]} -->

FedeDP commented:
I just rebased on top of latest master. :pray:

FedeDP commented:
`test-scap-{amd64,arm64}` tests are failing with:
> make[3]: *** No rule to make target 'protobuf-prefix/src/protobuf/target/lib/libprotobuf.a', needed by 'test/libscap/libscap_test'.  Stop.

Care to give it a look @Apteryks ?

FedeDP commented:
Moving to next milestone since we are nearing a release and this is not ready.
/milestone 0.19.0

FedeDP commented:
/milestone 0.20.0

Any news on this one?

Apteryks commented:
> `test-scap-{amd64,arm64}` tests are failing with:
> 
> > make[3]: *** No rule to make target 'protobuf-prefix/src/protobuf/target/lib/libprotobuf.a', needed by 'test/libscap/libscap_test'.  Stop.
> 
> Care to give it a look @Apteryks ?

Sorry for the long wait, I'll take a long at it now.

FedeDP commented:
No problem :)

Apteryks commented:
Lot's have changed (for the better it seems); there's less in this PR following a rebase.  I've been conservative in my edits in the hope that even the most exotic platforms shouldn't be impacted in the CI; let's see.

Apteryks commented:
> Lot's have changed (for the better it seems); there's less in this PR following a rebase. I've been conservative in my edits in the hope that even the most exotic platforms shouldn't be impacted in the CI; let's see.

Actually, the pkg-config files `Libs:` entries are back to contain a bit too much, which is correct for the static build case but leads to overlinking on GNU/Linux, which is not fatal. I'm not sure how to fix this cleanly for both builds to pass, so I guess I'll leave it at that.

Apteryks commented:
> Actually, the pkg-config files `Libs:` entries are back to contain a bit too much, which is correct for the static build case but leads to overlinking on GNU/Linux, which is not fatal. I'm not sure how to fix this cleanly for both builds to pass, so I guess I'll leave it at that.

The latest revision resolves that.  It's a bit hacky in the `add_pkgconfig_library` macro, but to be honest, it already was :-). The result looks great and I could build sysdig (which a few edits there to support the latest falcosecurity-libs).

I'm sharing the content of the freshly built pkg-config files, for a build configured with the flags: `cmake -DBUILD_SHARED_LIBS=ON -DUSE_BUNDLED_DEPS=OFF -DCMAKE_INSTALL_PREFIX=$PWD/install "-DBUILD_DRIVER=OFF" "-DENABLE_DKMS=OFF" "-DBUILD_LIBSCAP_MODERN_BPF=ON"`:

```
$ cat install/lib64/pkgconfig/libscap.pc
prefix=/home/maxim/src/falcosecurity-libs/buildc/install
libdir=${prefix}/lib64
includedir=${prefix}/include/falcosecurity

Name: libscap
Description: lib for System CAPture
Version: 0.21.0-19+d07beb8

# Note: jsoncpp and protobuf are required by scap_engine_gvisor, which
# currently lacks its own pkg-config file.
Requires: jsoncpp libpman protobuf
Requires.private: zlib
Libs: -L${libdir}  -lscap -lscap_event_schema -lscap_platform -lscap_engine_nodriver -lscap_engine_test_input -lscap_engine_source_plugin -lscap_engine_kmod -lscap_event_schema -lscap_platform -lscap_engine_bpf -lscap_event_schema -lscap_platform -lelf -lscap_engine_modern_bpf -lscap_engine_gvisor -lscap_event_schema
Cflags: -I${includedir} -I${includedir}/libscap -I${includedir}/driver -I/gnu/store/4b30xdqjj81dcyl5p518rp33vlm2iag8-profile/include

$ cat install/lib64/pkgconfig/libsinsp.pc
prefix=/home/maxim/src/falcosecurity-libs/buildc/install
libdir=${prefix}/lib64
includedir=${prefix}/include/falcosecurity

Name: libsinsp
Description: lib for System INSPection
Version: 0.21.0-19+d07beb8

Requires: libscap jsoncpp libcares gpr grpc grpc++ protobuf libcares libcrypto libssl
Requires.private: libcurl re2 tbb
Libs: -L${libdir}  -lsinsp
Cflags: -I${includedir} -I${includedir}/libsinsp -I${includedir}/driver -I/gnu/store/4b30xdqjj81dcyl5p518rp33vlm2iag8-profile/include/tbb -I/gnu/store/4b30xdqjj81dcyl5p518rp33vlm2iag8-profile/include

$ cat install/lib64/pkgconfig/libpman.pc
prefix=/home/maxim/src/falcosecurity-libs/buildc/install
libdir=${prefix}/lib64
includedir=${prefix}/include

Name: libpman
Description: Utility library for BPF probes
Version: 0.21.0-19+d07beb8

Requires: libbpf zlib
Libs: -L${libdir} -lpman -lscap_event_schema -lscap_platform
Cflags: -I${includedir}
```

Apteryks commented:
I think that's ready to be reviewed again.

Apteryks commented:
Only the sinsp example with pkg-config test appears to fail when building as a shared library on Linux and MacOS. I'll look into it; perhaps the install directory needs to be added to `PKG_CONFIG_PATH`.

Apteryks commented:
These tests pass locally, on my Guix System.  I need to use lib64 instead of lib in the PKG_CONFIG_PATH (and LIBRARY_PATH) though, so perhaps the paths in the ci.yml file need a similar adjustment?

Apteryks commented:
Phew.  I think my last push hopefully addresses the remaining test failures.  I've made the Requires and Requires.private for libscap conditionally computed like for libsinsp, and dropped what seemed like an extraneous protobuf link dependency I had added to the scap unit test.

Apteryks commented:
ugh. format code.  I thought I had cmake-format'ed everything already.

Apteryks commented:
OK, I've run `make format-all` and pushed; hopefully CI is green now.

FedeDP commented:
/cc @geraldcombs PTAL thanks!

Apteryks commented:
Hi; there was a single test error, apparently because it was cancelled: https://github.com/falcosecurity/libs/actions/runs/13001896582/job/36273360103  

Could it please be retried?

FedeDP commented:
Done!

Apteryks commented:
The single test failure looks like:

```
Run cd build/test/libsinsp_e2e/
./libsinsp_e2e_tests: 1: Syntax error: ";" unexpected
Error: Process completed with exit code 2.
```

That only happens on ARM; any ideas what is going on here?

FedeDP commented:
Never seen that; @therealbobo perhaps can help us?

therealbobo commented:
Never seen that, I'll build locally to see if the error pops up

geraldcombs commented:
I was able to test this with Stratoshark on macOS, and everything built and ran fine with no modifications required for the .pc files.

FedeDP commented:
Thanks Gerald!
@Apteryks can you rebase to fix the conflict? Then we are hopefully good to go :)

Apteryks commented:
> Thanks Gerald! @Apteryks can you rebase to fix the conflict? Then we are hopefully good to go :)

Done!

FedeDP reviewed (approved):
/approve

poiana commented:
LGTM label has been added.  <details>Git tree hash: 041b2a198f1e9774e2b4b16227a4e9654672524f</details>

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1842#" title="Author self-approved">Apteryks</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1842#pullrequestreview-2615580980" title="Approved">FedeDP</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [FedeDP]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->
